examples/gno.land/p/demo/varmeta_md/inline.gno

```diff
@@ -0,0 +1,1024 @@
+package varmeta_md
+
+import (
+	"bytes"
+	"regexp"
+	"strconv"
+)
+
+var (
+	urlRe    = `((https?|ftp):\/\/|\/)[-A-Za-z0-9+&@#\/%?=~_|!:,.;\(\)]+`
+	anchorRe = regexp.MustCompile(`^(<a\shref="` + urlRe + `"(\stitle="[^"<>]+")?\s?>` + urlRe + `<\/a>)`)
+)
+
+func (p *parser) inline(out *bytes.Buffer, data []byte) {
+	if p.nesting >= p.maxNesting {
+		return
+	}
+	p.nesting++
+
+	i, end := 0, 0
+	for i < len(data) {
+		for end < len(data) && p.inlineCallback[data[end]] == nil {
+			end++
+		}
+
+		p.r.NormalText(out, data[i:end])
+
+		if end >= len(data) {
+			break
+		}
+		i = end
+
+		handler := p.inlineCallback[data[end]]
+		if consumed := handler(p, out, data, i); consumed == 0 {
+			end = i + 1
+		} else {
+			i += consumed
+			end = i
+		}
+	}
+
+	p.nesting--
+}
+
+func emphasis(p *parser, out *bytes.Buffer, data []byte, offset int) int {
+	data = data[offset:]
+	c := data[0]
```

Comment:
Accessing `data[0]` when the data slice is empty can cause a runtime panic. I think need to check the length of data to prevent accessing an empty slice.
```suggestion
	data = data[offset:]
	if len(data) == 0 {
		return 0
	}
	c := data[0]
```

This pattern is also found in other functions (e.g., `codeSpan`). please check them just for safety.